mod/NomaiTextQoL.cs

```diff
@@ -1,42 +1,146 @@
-﻿using HarmonyLib;
+using HarmonyLib;
+using System.Linq;
+using System;
+using UnityEngine;
+using System.Collections.Generic;
 
 namespace ArchipelagoRandomizer
 {
     [HarmonyPatch]
     internal class NomaiTextQoL
     {
         public static bool AutoNomaiText;
+        public static bool ColorNomaiText = true;
         public static float TranslateTime = 0.2f;
 
+        public static Dictionary<string, Location> ManualScrollLocations = new()
+        {
+            { "Arc_BH_City_School_BigBangLesson", Location.BH_SOLANUM_REPORT },
+            { "Arc_TT_Tower_CT", Location.AT_HGT_TOWERS },
+            { "Arc_TT_Tower_BH_1", Location.AT_BH_TOWER }
+        };
+
         // Auto-expand all Nomai text in the game as a Quality of Life feature
         [HarmonyPostfix, HarmonyPatch(typeof(NomaiWallText), nameof(NomaiWallText.LateInitialize))]
         public static void NomaiWallText_LateInitialize_Postfix(NomaiWallText __instance)
         {
-            if (!AutoNomaiText) return;
-            foreach (NomaiTextLine child in __instance.GetComponentsInChildren<NomaiTextLine>())
+            if (ColorNomaiText)
             {
-                child._state = NomaiTextLine.VisualState.UNREAD;
+                // we can't directly get the logs connected to arcs,
+                // so we have to do this roundabout system (also used in base game)
+                // to iterate through a bunch of conditions, and find the arc that matches the condition of the log
+                for (int i = 0; i < __instance._listDBConditions.Count; i++)
+                {
+                    var nomaiTextData = __instance._listDBConditions[i];
```

Comment:
Should this be a `foreach`? I don't see `i` being used below.

Reply:
The `i` is used in the next line, and this is how the vanilla code is written. I want to make sure that I match the vanilla code exactly to be sure that I'm referencing the right arcs.

Reply:
Oh I didn't realize this was a copy-paste of a bunch of vanilla code. Matching its style is fine then.